src/app/core/auth/auth-http.service.ts

```diff
@@ -1,52 +1,45 @@
 import {Injectable} from "@angular/core";
 import {AuthService} from "./auth.service";
-import {Http, RequestOptionsArgs, Response, ResponseContentType} from "@angular/http";
+import {ResponseContentType} from "@angular/http";
 import {Observable} from "rxjs";
 import 'rxjs/add/operator/map';
+import {HttpClient, HttpErrorResponse,HttpResponse} from "@angular/common/http";
 
 @Injectable()
 export class AuthHttp {
 
-  constructor(private authService: AuthService, private http: Http) {
+  constructor(private authService: AuthService, private http: HttpClient) {
   }
 
-  public get(url: string, options?: RequestOptionsArgs): Observable<Response> {
+  public get(url: string, options?:any): Observable<any> {
```

Comment:
Maybe it would be better return `Observable<Object>`?

Or `Observable<HttpResponse<T>>`,  `Observable<T>`
```
get<T>(url: string, options?:any): Observable<T> {
  return this.http.get<T>(url, options).catch(error => this.processError(error));
}
```